storage/src/main/java/tech/pegasys/teku/storage/server/metadata/V6DatabaseMetadata.java

```diff
@@ -0,0 +1,160 @@
+ * be loaded providing a simple way to experiment with different values without it being fixed at
+ * database creation.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class V6DatabaseMetadata {
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class SingleDBMetadata {
+    @JsonProperty("configuration")
+    private RocksDbConfiguration configuration;
+
+    public SingleDBMetadata() {}
+
+    public SingleDBMetadata(RocksDbConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    public RocksDbConfiguration getConfiguration() {
+      return configuration;
+    }
+
+    @Override
+    public String toString() {
+      return "SingleDBMetadata{" + "configuration=" + configuration + '}';
+    }
+  }
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class SeparateDBMetadata {
+    @JsonProperty("hotDbConfiguration")
+    private RocksDbConfiguration hotDbConfiguration;
+
+    @JsonProperty("archiveDbConfiguration")
+    private RocksDbConfiguration archiveDbConfiguration;
+
+    public SeparateDBMetadata() {}
+
+    public SeparateDBMetadata(
+        RocksDbConfiguration hotDbConfiguration, RocksDbConfiguration archiveDbConfiguration) {
+      this.hotDbConfiguration = hotDbConfiguration;
+      this.archiveDbConfiguration = archiveDbConfiguration;
+    }
+
+    public RocksDbConfiguration getHotDbConfiguration() {
+      return hotDbConfiguration;
+    }
+
+    public RocksDbConfiguration getArchiveDbConfiguration() {
+      return archiveDbConfiguration;
+    }
+
+    @Override
+    public String toString() {
+      return "SeparateDBMetadata{"
+          + "hotDbConfiguration="
+          + hotDbConfiguration
+          + ", archiveDbConfiguration="
+          + archiveDbConfiguration
+          + '}';
+    }
+  }
+
+  @JsonProperty("singleDb")
+  private SingleDBMetadata singleDb;
+
+  @JsonProperty("separateDb")
+  private SeparateDBMetadata separateDb;
+
+  public V6DatabaseMetadata() {}
+
+  private V6DatabaseMetadata(RocksDbConfiguration singleDbConfiguration) {
+    this.singleDb = new SingleDBMetadata(singleDbConfiguration);
+  }
+
+  private V6DatabaseMetadata(
+      RocksDbConfiguration hotDbConfiguration, RocksDbConfiguration archiveDbConfiguration) {
+    this.separateDb = new SeparateDBMetadata(hotDbConfiguration, archiveDbConfiguration);
+  }
+
+  public static V6DatabaseMetadata singleDBDefault() {
+    return new V6DatabaseMetadata(RocksDbConfiguration.v6SingleDefaults());
+  }
+
+  public static V6DatabaseMetadata separateDBDefault() {
+    return new V6DatabaseMetadata(
+        RocksDbConfiguration.v5HotDefaults(), RocksDbConfiguration.v5ArchiveDefaults());
+  }
+
+  public Optional<SingleDBMetadata> getSingleDbConfiguration() {
+    return Optional.ofNullable(singleDb);
+  }
+
+  public Optional<SeparateDBMetadata> getSeparateDbConfiguration() {
+    return Optional.ofNullable(separateDb);
+  }
```

Comment:
Looks like these are being serialized - there's probably a flag to turn off this behavior

Reply:
Oh, thanks for pointing this 👍 
Fix here: https://github.com/PegaSysEng/teku/pull/2934/commits/1b023b894453f925fb7632eeb8d99d6486202bf8
Also excluded serialization of a null field